packages/utils/src/dag.ts

```diff
@@ -42,7 +42,7 @@ export default class DAG {
     if (!node) {
       throw new Error(`Node ${id} not found`)
     }
-    return this.layers[node.layer - 1]
+    return this.layers[node.layer - 1] || 0
```

Comment:
Why we need or 0 here?

Reply:
In case that the layer doesn't exist. It happens in a previous situation which I fixed.